Formula/mold.rb

```diff
@@ -62,16 +62,16 @@ def install
     system "cmake", "--build", "build"
     system "cmake", "--install", "build"
 
-    inreplace buildpath.glob("test/macho/*.sh"), "./ld64", bin/"ld64.mold", false
-    inreplace buildpath.glob("test/elf/*.sh") do |s|
-      s.gsub!(%r{(\./|`pwd`/)?mold-wrapper}, lib/"mold/mold-wrapper", false)
-      s.gsub!(%r{(\.|`pwd`)/mold}, bin/"mold", false)
-      s.gsub!(/-B(\.|`pwd`)/, "-B#{libexec}/mold", false)
-    end
     pkgshare.install "test"
   end
 
   test do
+    assert_match "mold #{version}", shell_output("#{bin}/mold -v")
+
+    # Support for Mach-O targets was removed from mold in v1.8.0.
+    # https://github.com/rui314/mold/releases/tag/v1.8.0
+    return if OS.mac?
```

Comment:
Should we simply mark the formula linux-only?

Reply:
I wonder if it could be kept, for the same reason why formulae such as `x86_64-elf-binutils` and `x86_64-elf-gdb` are available on macOS. It builds on macOS, and should continue to function as an ELF linker (which will supposedly remain the case, as mold is the subset of sold, the commercially licensed version with Mach-O support).

Cc: @p-linnane

Reply:
I will defer to @chenrui333 on this one.

Reply:
It works on macOS (though with now reduced functionality), so it seems odd to make it Linux-only.

Reply:
That said, we should try to link something trivial and see if it produces an ELF executable.

Reply:
> That said, we should try to link something trivial and see if it produces an ELF executable.

Makes sense. Please have a look at the new changes.

Also, I'm thinking of adding a caveat to notify the existing users. Something like:

```plaintext
macOS support has been removed from mold since version 1.8.0.
See https://github.com/bluewhalesystems/sold for details.
```

Any opinions?

Reply:
yeah, maybe caveat would be the better way.